simple-wp-helpdesk/frontend/class-portal.php

```
@@ -500,6 +465,316 @@ function swh_render_portal_no_token() {
+	$post     = $ticket_id ? get_post( $ticket_id ) : null;
+	$db_token = $ticket_id ? swh_get_string_meta( $ticket_id, '_ticket_token' ) : '';
+	if ( ! $post || '' === $db_token || '' === $token || ! hash_equals( $db_token, $token ) ) {
+		wp_die( esc_html__( 'Invalid request.', 'simple-wp-helpdesk' ), 403 );
+	}
+	$portal_url = swh_get_secure_ticket_link( $ticket_id );
+	if ( $portal_url ) {
+		wp_safe_redirect( $portal_url );
+		exit;
+	}
+	wp_die( esc_html__( 'Invalid request.', 'simple-wp-helpdesk' ), 403 );
+}
```

PHPMD (2.15.0)

[warning] The function swh_handle_portal_v2_csat() has a Cyclomatic Complexity of 12. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)